src/components/ConfirmedRoute.tsx

```diff
@@ -90,19 +79,21 @@ function ConfirmedRoute({mapboxAccessToken, transaction}) {
 
     useEffect(() => {
         MapboxToken.init();
-        return MapboxToken.stop;
+        return () => {
+            MapboxToken.stop();
+        };
     }, []);
 
     return (
         <>
-            {!isOffline && Boolean(mapboxAccessToken.token) ? (
+            {!isOffline && Boolean(mapboxAccessToken?.token) ? (
                 <DistanceMapView
-                    accessToken={mapboxAccessToken.token}
+                    accessToken={mapboxAccessToken?.token ?? ''}
                     mapPadding={CONST.MAP_PADDING}
                     pitchEnabled={false}
                     initialState={{
                         zoom: CONST.MAPBOX.DEFAULT_ZOOM,
-                        location: lodashGet(waypointMarkers, [0, 'coordinate'], CONST.MAPBOX.DEFAULT_COORDINATE),
+                        location: (waypointMarkers[0]?.coordinate ?? CONST.MAPBOX.DEFAULT_COORDINATE) as number[],
```

Comment:
Two things, add ?. before array index access + a little better assertion:

```suggestion
                        location: waypointMarkers?.[0]?.coordinate ?? (CONST.MAPBOX.DEFAULT_COORDINATE as [number, number]),

```